PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,367 @@
+            self.canvas.coords(drawItem.display_val, *self.cellCenter(i))
+        self.window.update()
+
+    def cleanUp(self, *args, **kwargs): # Customize clean up for sorting
+        super().cleanUp(*args, **kwargs) # Do the VisualizationApp clean up
+        self.fixCells()       # Restore cells to their coordinates in array
+
+    def makeButtons(self):
+        vcmd = (self.window.register(numericValidate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        newSizeArrayButton = self.addOperation(
+            "New", lambda: self.clickNew(), numArguments=1, validationCmd=vcmd)
+        insertButton = self.addOperation(
+            "Insert", lambda: self.clickInsert(), numArguments=1, validationCmd=vcmd)
+        deleteButton = self.addOperation(
+            "Remove", lambda: self.clickRemove(), numArguments=0)
+        peekButton = self.addOperation(
+            "Peek", lambda: self.clickPeek(), numArguments=0)
+        self.addAnimationButtons()
+        return [insertButton, deleteButton, peekButton]
+
+    def validArgument(self):
+        entered_text = self.getArgument()
+        if entered_text and entered_text.isdigit():
+            val = int(entered_text)
+            if val < 100:
+                return val
+    
+    def clickInsert(self):
+        val = self.validArgument()
+        if val is None:
+            self.setMessage("Input value must be an integer from 0 to 99.")
+        else:
+            if self.nItems >= self.size:
+                self.setMessage("Error! Queue is already full.")
+            else:
+                self.insert(val)
+                self.setMessage("Value {} inserted".format(val))
+        self.clearArgument()
+    
+    def clickRemove(self):
+        result = self.remove()
+        self.clearArgument()
+
+    def clickPeek(self):
+        if self.nItems <= 0: self.setMessage("Error! Queue is empty.")
+        else:
+            self.peek()
+            self.setMessage("Value at front is {}".format(self.list[self.nItems-1].val))
+
+    def clickNew(self):
+        val = self.validArgument()
+        # If the number of cells desired wouldn't fit on the screen, error message
+        if val is None or self.window.winfo_width() <= self.ARRAY_X0 + (val+1) * self.CELL_SIZE:
+            self.setMessage("This array size is too big to display")
+        elif val == 0:
+            self.setMessage("This array size is too small")
+        else:
+            self.newArraySize(val)
+        self.clearArgument()
+
+    def enableButtons(self, enable=True):
+        for btn in self.buttons:
+            btn.config(state=NORMAL if enable else DISABLED)
+    
+    def startAnimations(self):
+        self.enableButtons(enable=False)
+        super().startAnimations()
+            
+    def stopAnimations(self):
+        super().stopAnimations()
+        self.enableButtons(enable=True)
+        self.argumentChanged()
+
```

Comment:
You don't need to override these three methods from VisualizationApp.  They already handle the enabling/disabling of operations buttons, stopping animations, and checking if arguments have changed.  Just remove lines 351 - 363.